### PR TITLE
Keycloak

### PR DESCRIPTION
**Pull request by cehbrecht**
This PR adds support for an external Keycloak authentication service using OAuth2 tokens. Currently the OAuth *client credentials* workflow is described and supported.

Changes:
* support for token validation with keycloak
* docs and the overview image are updated.
* added an example notebook with a keycloak demo.
* OWS client registration is using now basic authentication.

For testing you need a keycloak instance which you can setup locally with Ansible:
https://github.com/bird-house/ansible-keycloak-playbook

### CODE REVIEW

**Comment by cehbrecht** [comment]
The keycloak demo with also tested at CMCC with Ophidia by @minomicetto.

@minomicetto Please note that I have opened a new repo for the keycloak notebook (master branch):
https://github.com/bird-house/ansible-keycloak-playbook

**Comment by cehbrecht** [comment]
@plogerais @jasonb5 You are welcome to add comments if you like.

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -171,7 +171,43 @@ Let's say your proxy certificate is ``cert.pem``, then run the exceute request a
...
+Try the demo notebook to access a token from the keycloak and execute a WPS process.
+
+Use ``client_id=demo`` and copy the client secret from Keycloak in `Clients/demo/Credentials/Secret` (see screenshot).
+
+.. image:: _images/keycloak-client-secret.png
+
+.. toctree::
+   :maxdepth: 1
+
+   notebooks/twitcher-keycloak-demo
```
Comment (tlvu) [inline docs/source/tutorial.rst]: That section did not display correctly, missing `.ipynb` extension?

2019-11-13-135928_1005x877_scrot

Comment (cehbrecht) [reply docs/source/tutorial.rst]: Needs `nbsphinx` to render. I have updated the readthedocs config ... hopefully this works.

**Comment by cehbrecht** [comment]
@tlvu @fmigneault thanks for review.

**Pull request merged**